## Final Project Submission

Please fill out: 
* Student name: Willian Nyawir
* Student pace: self paced / part time / full time
* Scheduled project review date/time: Friday 17th Dec 2026
* Instructor name: Mrs Mwikali
* Blog post URL:


# ANALYSIS OF THE DATASET

# BUSINESS PROBLEM
The company plans to expand into the aviation industry but lacks knowledge about the safety risks associated with different aircraft. This project analyzes historical aviation accident data to identify aircraft makes and models with lower risk profiles, enabling data-driven decisions on which aircraft to purchase for commercial and private operations.


# DATA UNDERSTANDING

In [76]:
#Importing necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
df=pd.read_csv('Aviation_Data.csv')
df

C:\Users\user\AppData\Local\Temp\ipykernel_20772\2010058794.py:5: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  df=pd.read_csv('Aviation_Data.csv')


Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country   Latitude  Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1       BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2        Saltville, VA  United States  36.922223 -81.878056          NaN   
3           EUREKA, CA  United States        NaN        NaN          NaN   
4           Canton, OH  United States        NaN        NaN          NaN   
...                ...            ...        ...        ...          ...   
90343    Annapolis, MD  United States        NaN        NaN          NaN   
90344      Hampton, NH  United States        NaN        NaN          NaN   
90345       Payson, AZ  United States    341525N   1112021W          PAN   
90346       Morgan, UT  United States        NaN        NaN          NaN   
90347       Athens, GA  United States        NaN        NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    NaN                  NaN   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  NaN   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 NaN               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               N

In [77]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90348 entries, 0 to 90347
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      90348 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50132 non-null  object 
 9   Airport.Name            52704 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87507 non-null  object 
 14  Make                    88826 non-null

In [78]:
df.describe() #gives statistical summary of numerical columns

Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       82805.000000          77488.000000            76379.000000   
mean            1.146585              0.647855                0.279881   
std             0.446510              5.485960                1.544084   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured  
count          76956.000000     82977.000000  
mean               0.357061         5.325440  
std                2.235625        27.913634  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         1.000000  
75%                0.000000         2.000000  
max              380.000000       699.000000

In [79]:
df.columns #lists all column names

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

In [80]:
df.shape #calculates number of rows and columns

(90348, 31)

In [81]:
df.ndim #checks number of dimensions

2

The above data has 90348 rows and 31 columns. It also has 2 dimensions. The statistical report of the numerical data is also shown above

## DATA CLEANING

In [82]:
used_columns = df[['Make','Model','Injury.Severity','Aircraft.damage','Aircraft.Category',
                   'Number.of.Engines','Publication.Date','Engine.Type','Purpose.of.flight',
                   'Weather.Condition','Total.Fatal.Injuries',
                   'Total.Serious.Injuries','Total.Minor.Injuries','Registration.Number','Investigation.Type']]

#It organizes the column titles by removing spaces, converting to lowercase and replacing periods with underscores
used_columns.columns = (
    used_columns.columns
    .str.strip()
    .str.lower()
    .str.replace('.', '_', regex=False)
)
used_columns

make      model injury_severity aircraft_damage  \
0                         Stinson      108-3        Fatal(2)       Destroyed   
1                           Piper   PA24-180        Fatal(4)       Destroyed   
2                          Cessna       172M        Fatal(3)       Destroyed   
3                        Rockwell        112        Fatal(2)       Destroyed   
4                          Cessna        501        Fatal(1)       Destroyed   
...                           ...        ...             ...             ...   
90343                       PIPER  PA-28-151           Minor             NaN   
90344                    BELLANCA       7ECA             NaN             NaN   
90345  AMERICAN CHAMPION AIRCRAFT      8GCBC       Non-Fatal     Substantial   
90346                      CESSNA       210N             NaN             NaN   
90347                       PIPER  PA-24-260           Minor             NaN   

      aircraft_category  number_of_engines publication_date    engine_type  \
0                   NaN                1.0              NaN  Reciprocating   
1                   NaN                1.0       19-09-1996  Reciprocating   
2                   NaN                1.0       26-02-2007  Reciprocating   
3                   NaN                1.0       12-09-2000  Reciprocating   
4                   NaN                NaN       16-04-1980            NaN   
...                 ...                ...              ...            ...   
90343               NaN                NaN       29-12-2022            NaN   
90344               NaN                NaN              NaN            NaN   
90345          Airplane                1.0       27-12-2022            NaN   
90346               NaN                NaN              NaN            NaN   
90347               NaN                NaN       30-12-2022            NaN   

      purpose_of_flight weather_condition  total_fatal_injuries  \
0              Personal               UNK                   2.0   
1              Personal               UNK                   4.0   
2              Personal               IMC                   3.0   
3              Personal               IMC                   2.0   
4              Personal               VMC                   1.0   
...                 ...               ...                   ...   
90343          Personal               NaN                   0.0   
90344               NaN               NaN                   0.0   
90345          Personal               VMC                   0.0   
90346          Personal               NaN                   0.0   
90347          Personal               NaN                   0.0   

       total_serious_injuries  total_minor_injuries registration_number  \
0                         0.0                   0.0              NC6404   
1                         0.0                   0.0              N5069P   
2                         NaN                   NaN              N5142R   
3                         0.0                   0.0              N1168J   
4                         2.0                   NaN               N15NY   
...                       ...                   ...                 ...   
90343                     1.0                   0.0              N1867H   
90344                     0.0                   0.0              N2895Z   
90345                     0.0                   0.0              N749PJ   
90346                     0.0                   0.0              N210CU   
90347                     1.0                   0.0              N9026P   

      investigation_type  
0               Accident  
1               Accident  
2               Accident  
3               Accident  
4               Accident  
...                  ...  
90343           Accident  
90344           Accident  
90345           Accident  
90346           Accident  
90347           Accident  

[90348 rows x 15 columns]

In [83]:
used_columns.isna().sum() #counts no. of null values


make                       1522
model                      1551
injury_severity            2459
aircraft_damage            4653
aircraft_category         58061
number_of_engines          7543
publication_date          16689
engine_type                8555
purpose_of_flight          7651
weather_condition          5951
total_fatal_injuries      12860
total_serious_injuries    13969
total_minor_injuries      13392
registration_number        2841
investigation_type            0
dtype: int64

In [84]:
used_columns=used_columns.dropna() #removes null values
used_columns

make   model injury_severity aircraft_damage  \
7                 Cessna     140       Non-Fatal     Substantial   
8                 Cessna    401B       Non-Fatal     Substantial   
12              Bellanca  17-30A       Non-Fatal       Destroyed   
13                Cessna   R172K        Fatal(1)       Destroyed   
14                Navion       A        Fatal(1)       Destroyed   
...                  ...     ...             ...             ...   
90098             CESSNA     150       Non-Fatal     Substantial   
90106             CESSNA   177RG       Non-Fatal     Substantial   
90120              BEECH    B-60       Non-Fatal     Substantial   
90194  STEPHEN J HOFFMAN  MS-500           Minor     Substantial   
90226           LUSCOMBE      8E       Non-Fatal     Substantial   

      aircraft_category  number_of_engines publication_date    engine_type  \
7              Airplane                1.0       01-01-1982  Reciprocating   
8              Airplane                2.0       01-01-1982  Reciprocating   
12             Airplane                1.0       02-01-1983  Reciprocating   
13             Airplane                1.0       02-01-1983  Reciprocating   
14             Airplane                1.0       02-01-1983  Reciprocating   
...                 ...                ...              ...            ...   
90098          Airplane                1.0       20-12-2022  Reciprocating   
90106          Airplane                1.0       20-12-2022  Reciprocating   
90120          Airplane                2.0       20-12-2022  Reciprocating   
90194          Airplane                1.0       20-12-2022  Reciprocating   
90226          Airplane                1.0       20-12-2022  Reciprocating   

      purpose_of_flight weather_condition  total_fatal_injuries  \
7              Personal               VMC                   0.0   
8              Business               IMC                   0.0   
12             Personal               IMC                   0.0   
13             Personal               IMC                   1.0   
14             Personal               IMC                   1.0   
...                 ...               ...                   ...   
90098          Personal               VMC                   0.0   
90106          Personal               VMC                   0.0   
90120          Personal               VMC                   0.0   
90194              ASHO               VMC                   0.0   
90226          Personal               VMC                   0.0   

       total_serious_injuries  total_minor_injuries registration_number  \
7                         0.0                   0.0              N2482N   
8                         0.0                   0.0              N7967Q   
12                        0.0                   1.0              N14779   
13                        0.0                   0.0              N758SK   
14                        0.0                   0.0              N4876K   
...                       ...                   ...                 ...   
90098                     0.0                   0.0              N18158   
90106                     0.0                   0.0              N2047Q   
90120                     0.0                   0.0              N2889W   
90194                     1.0                   0.0              N778MS   
90226                     0.0                   0.0              N1855K   

      investigation_type  
7               Accident  
8               Accident  
12              Accident  
13              Accident  
14              Accident  
...                  ...  
90098           Accident  
90106           Accident  
90120           Accident  
90194           Accident  
90226           Accident  

[19073 rows x 15 columns]

In [85]:
#only has aircraft_category as airplane 



airplane_dataset=used_columns.loc[used_columns['aircraft_category']=='Airplane']
airplane_dataset

make   model injury_severity aircraft_damage  \
7                 Cessna     140       Non-Fatal     Substantial   
8                 Cessna    401B       Non-Fatal     Substantial   
12              Bellanca  17-30A       Non-Fatal       Destroyed   
13                Cessna   R172K        Fatal(1)       Destroyed   
14                Navion       A        Fatal(1)       Destroyed   
...                  ...     ...             ...             ...   
90098             CESSNA     150       Non-Fatal     Substantial   
90106             CESSNA   177RG       Non-Fatal     Substantial   
90120              BEECH    B-60       Non-Fatal     Substantial   
90194  STEPHEN J HOFFMAN  MS-500           Minor     Substantial   
90226           LUSCOMBE      8E       Non-Fatal     Substantial   

      aircraft_category  number_of_engines publication_date    engine_type  \
7              Airplane                1.0       01-01-1982  Reciprocating   
8              Airplane                2.0       01-01-1982  Reciprocating   
12             Airplane                1.0       02-01-1983  Reciprocating   
13             Airplane                1.0       02-01-1983  Reciprocating   
14             Airplane                1.0       02-01-1983  Reciprocating   
...                 ...                ...              ...            ...   
90098          Airplane                1.0       20-12-2022  Reciprocating   
90106          Airplane                1.0       20-12-2022  Reciprocating   
90120          Airplane                2.0       20-12-2022  Reciprocating   
90194          Airplane                1.0       20-12-2022  Reciprocating   
90226          Airplane                1.0       20-12-2022  Reciprocating   

      purpose_of_flight weather_condition  total_fatal_injuries  \
7              Personal               VMC                   0.0   
8              Business               IMC                   0.0   
12             Personal               IMC                   0.0   
13             Personal               IMC                   1.0   
14             Personal               IMC                   1.0   
...                 ...               ...                   ...   
90098          Personal               VMC                   0.0   
90106          Personal               VMC                   0.0   
90120          Personal               VMC                   0.0   
90194              ASHO               VMC                   0.0   
90226          Personal               VMC                   0.0   

       total_serious_injuries  total_minor_injuries registration_number  \
7                         0.0                   0.0              N2482N   
8                         0.0                   0.0              N7967Q   
12                        0.0                   1.0              N14779   
13                        0.0                   0.0              N758SK   
14                        0.0                   0.0              N4876K   
...                       ...                   ...                 ...   
90098                     0.0                   0.0              N18158   
90106                     0.0                   0.0              N2047Q   
90120                     0.0                   0.0              N2889W   
90194                     1.0                   0.0              N778MS   
90226                     0.0                   0.0              N1855K   

      investigation_type  
7               Accident  
8               Accident  
12              Accident  
13              Accident  
14              Accident  
...                  ...  
90098           Accident  
90106           Accident  
90120           Accident  
90194           Accident  
90226           Accident  

[16784 rows x 15 columns]

# DATA MANIPULATION


In [86]:
#Standardize the 'make' column by stripping whitespace and converting to uppercase
used_columns['make'] = (
    used_columns['make']
    .str.strip()
    .str.upper()
)

used_columns

C:\Users\user\AppData\Local\Temp\ipykernel_20772\622886935.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  used_columns['make'] = (


make   model injury_severity aircraft_damage  \
7                 CESSNA     140       Non-Fatal     Substantial   
8                 CESSNA    401B       Non-Fatal     Substantial   
12              BELLANCA  17-30A       Non-Fatal       Destroyed   
13                CESSNA   R172K        Fatal(1)       Destroyed   
14                NAVION       A        Fatal(1)       Destroyed   
...                  ...     ...             ...             ...   
90098             CESSNA     150       Non-Fatal     Substantial   
90106             CESSNA   177RG       Non-Fatal     Substantial   
90120              BEECH    B-60       Non-Fatal     Substantial   
90194  STEPHEN J HOFFMAN  MS-500           Minor     Substantial   
90226           LUSCOMBE      8E       Non-Fatal     Substantial   

      aircraft_category  number_of_engines publication_date    engine_type  \
7              Airplane                1.0       01-01-1982  Reciprocating   
8              Airplane                2.0       01-01-1982  Reciprocating   
12             Airplane                1.0       02-01-1983  Reciprocating   
13             Airplane                1.0       02-01-1983  Reciprocating   
14             Airplane                1.0       02-01-1983  Reciprocating   
...                 ...                ...              ...            ...   
90098          Airplane                1.0       20-12-2022  Reciprocating   
90106          Airplane                1.0       20-12-2022  Reciprocating   
90120          Airplane                2.0       20-12-2022  Reciprocating   
90194          Airplane                1.0       20-12-2022  Reciprocating   
90226          Airplane                1.0       20-12-2022  Reciprocating   

      purpose_of_flight weather_condition  total_fatal_injuries  \
7              Personal               VMC                   0.0   
8              Business               IMC                   0.0   
12             Personal               IMC                   0.0   
13             Personal               IMC                   1.0   
14             Personal               IMC                   1.0   
...                 ...               ...                   ...   
90098          Personal               VMC                   0.0   
90106          Personal               VMC                   0.0   
90120          Personal               VMC                   0.0   
90194              ASHO               VMC                   0.0   
90226          Personal               VMC                   0.0   

       total_serious_injuries  total_minor_injuries registration_number  \
7                         0.0                   0.0              N2482N   
8                         0.0                   0.0              N7967Q   
12                        0.0                   1.0              N14779   
13                        0.0                   0.0              N758SK   
14                        0.0                   0.0              N4876K   
...                       ...                   ...                 ...   
90098                     0.0                   0.0              N18158   
90106                     0.0                   0.0              N2047Q   
90120                     0.0                   0.0              N2889W   
90194                     1.0                   0.0              N778MS   
90226                     0.0                   0.0              N1855K   

      investigation_type  
7               Accident  
8               Accident  
12              Accident  
13              Accident  
14              Accident  
...                  ...  
90098           Accident  
90106           Accident  
90120           Accident  
90194           Accident  
90226           Accident  

[19073 rows x 15 columns]

In [87]:
#make date to be for the past 20 years only
used_columns['publication_date'] = pd.to_datetime(used_columns['publication_date'], errors='coerce')
used_columns = used_columns[used_columns['publication_date'] >= '2004-01-01']
used_columns

C:\Users\user\AppData\Local\Temp\ipykernel_20772\1410114761.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  used_columns['publication_date'] = pd.to_datetime(used_columns['publication_date'], errors='coerce')


make              model injury_severity  \
11898                         CESSNA                152        Fatal(1)   
11903                       SIKORSKY               S76A        Fatal(1)   
25766  CONSOLIDATED AERONAUTICS INC.           LA-4-200        Fatal(1)   
35750                   AEROSPATIALE             SA316B       Non-Fatal   
36335                        B??LKOW            BO-105S        Fatal(2)   
...                              ...                ...             ...   
89901     WACO CLASSIC AIRCRAFT CORP            2T-1A-2           Minor   
89908                        STINSON                10A       Non-Fatal   
89933                         CESSNA                120           Minor   
89941                         CESSNA                152       Non-Fatal   
89950                          PIPER  PA-18-105 SPECIAL       Non-Fatal   

      aircraft_damage aircraft_category  number_of_engines publication_date  \
11898       Destroyed          Airplane                1.0       2013-08-04   
11903     Substantial        Helicopter                2.0       2013-08-04   
25766     Substantial          Airplane                1.0       2016-01-02   
35750     Substantial        Helicopter                1.0       2009-02-02   
36335       Destroyed        Helicopter                2.0       2009-02-02   
...               ...               ...                ...              ...   
89901     Substantial          Airplane                1.0       2022-01-11   
89908     Substantial          Airplane                1.0       2022-08-12   
89933     Substantial          Airplane                1.0       2022-08-12   
89941     Substantial          Airplane                1.0       2022-08-12   
89950     Substantial          Airplane                1.0       2022-08-12   

         engine_type          purpose_of_flight weather_condition  \
11898  Reciprocating              Instructional               VMC   
11903    Turbo Shaft        Executive/corporate               VMC   
25766  Reciprocating                   Personal               VMC   
35750    Turbo Shaft                Positioning               VMC   
36335    Turbo Shaft                Positioning               VMC   
...              ...                        ...               ...   
89901  Reciprocating                   Personal               VMC   
89908  Reciprocating                   Personal               VMC   
89933  Reciprocating                   Personal               VMC   
89941  Reciprocating              Instructional               VMC   
89950  Reciprocating  Public Aircraft - Federal               VMC   

       total_fatal_injuries  total_serious_injuries  total_minor_injuries  \
11898                   1.0                     1.0                   4.0   
11903                   1.0                     1.0                   4.0   
25766                   1.0                     1.0                   1.0   
35750                   0.0                     0.0                   3.0   
36335                   2.0                     1.0                   0.0   
...                     ...                     ...                   ...   
89901                   0.0                     2.0                   0.0   
89908                   0.0                     0.0                   0.0   
89933                   0.0                     1.0                   0.0   
89941                   0.0                     0.0                   0.0   
89950                   0.0                     0.0                   0.0   

      registration_number investigation_type  
11898              N4956B           Accident  
11903              N176FJ           Accident  
25766               N76KK           Accident  
35750              N981CF           Accident  
36335              N911LF           Accident  
...                   ...                ...  
89901               N21GL           Accident  
89908              N5847X           Accident  
89933

In [88]:
#Finding data that only has investigation type as Accident
accident_data=airplane_dataset.loc[airplane_dataset['investigation_type']=='Accident']
accident_data

make   model injury_severity aircraft_damage  \
7                 Cessna     140       Non-Fatal     Substantial   
8                 Cessna    401B       Non-Fatal     Substantial   
12              Bellanca  17-30A       Non-Fatal       Destroyed   
13                Cessna   R172K        Fatal(1)       Destroyed   
14                Navion       A        Fatal(1)       Destroyed   
...                  ...     ...             ...             ...   
90098             CESSNA     150       Non-Fatal     Substantial   
90106             CESSNA   177RG       Non-Fatal     Substantial   
90120              BEECH    B-60       Non-Fatal     Substantial   
90194  STEPHEN J HOFFMAN  MS-500           Minor     Substantial   
90226           LUSCOMBE      8E       Non-Fatal     Substantial   

      aircraft_category  number_of_engines publication_date    engine_type  \
7              Airplane                1.0       01-01-1982  Reciprocating   
8              Airplane                2.0       01-01-1982  Reciprocating   
12             Airplane                1.0       02-01-1983  Reciprocating   
13             Airplane                1.0       02-01-1983  Reciprocating   
14             Airplane                1.0       02-01-1983  Reciprocating   
...                 ...                ...              ...            ...   
90098          Airplane                1.0       20-12-2022  Reciprocating   
90106          Airplane                1.0       20-12-2022  Reciprocating   
90120          Airplane                2.0       20-12-2022  Reciprocating   
90194          Airplane                1.0       20-12-2022  Reciprocating   
90226          Airplane                1.0       20-12-2022  Reciprocating   

      purpose_of_flight weather_condition  total_fatal_injuries  \
7              Personal               VMC                   0.0   
8              Business               IMC                   0.0   
12             Personal               IMC                   0.0   
13             Personal               IMC                   1.0   
14             Personal               IMC                   1.0   
...                 ...               ...                   ...   
90098          Personal               VMC                   0.0   
90106          Personal               VMC                   0.0   
90120          Personal               VMC                   0.0   
90194              ASHO               VMC                   0.0   
90226          Personal               VMC                   0.0   

       total_serious_injuries  total_minor_injuries registration_number  \
7                         0.0                   0.0              N2482N   
8                         0.0                   0.0              N7967Q   
12                        0.0                   1.0              N14779   
13                        0.0                   0.0              N758SK   
14                        0.0                   0.0              N4876K   
...                       ...                   ...                 ...   
90098                     0.0                   0.0              N18158   
90106                     0.0                   0.0              N2047Q   
90120                     0.0                   0.0              N2889W   
90194                     1.0                   0.0              N778MS   
90226                     0.0                   0.0              N1855K   

      investigation_type  
7               Accident  
8               Accident  
12              Accident  
13              Accident  
14              Accident  
...                  ...  
90098           Accident  
90106           Accident  
90120           Accident  
90194           Accident  
90226           Accident  

[16588 rows x 15 columns]

In [89]:
#Shows how often an accident for each make
accident_frequency = used_columns.groupby(['make', 'model']).size().reset_index(name='accident_count')
accident_frequency

make          model  accident_count
0    AERO COMMANDER            500               1
1    AERO COMMANDER    CALLAIR A 9               1
2           AERONCA           11BC               1
3           AERONCA            7AC               3
4           AERONCA            7DC               1
..              ...            ...             ...
457        YAKOVLEV         Yak 52               1
458          ZENAIR          CH601               1
459          ZENITH          CH601               1
460          ZENITH     STOL CH701               1
461          ZENITH  ZODIAC CH 650               1

[462 rows x 3 columns]

In [ ]:
#makes with most accidents
top_10_make = (
    used_columns['make']
    .value_counts()
    .head(10)
)
top_10_make


make
CESSNA          201
PIPER           104
BEECH            44
BELL             28
ROBINSON         21
EUROCOPTER       11
MOONEY           11
VANS              8
AEROSPATIALE      8
CHAMPION          8
Name: count, dtype: int64

In [91]:
#model of aircraft with most accidents
make_model_accidents = (
    used_columns
    .groupby(['make', 'model'])
    .size()
    .reset_index(name='Total_Accidents')
    .sort_values('Total_Accidents', ascending=False)
)

make_model_accidents.head(10)

make      model  Total_Accidents
114    CESSNA        172               32
130    CESSNA        182               15
369     PIPER       PA28               12
346     PIPER  PA-18-150               11
389  ROBINSON        R22               10
106    CESSNA        150               10
365     PIPER       PA18                9
366     PIPER       PA22                7
392  ROBINSON        R44                7
190  CHAMPION      7GCBC                6

In [100]:
#level of fatality after accident based on make 
fatality_analysis = (
    used_columns
    .groupby(['make', 'model'])
    .agg({
        'total_fatal_injuries': 'sum',
        'total_serious_injuries': 'sum',
        'total_minor_injuries': 'sum'
    })
    .reset_index()
    .sort_values('total_fatal_injuries', ascending=False)
)
fatality_analysis.head(10)

make model  total_fatal_injuries  total_serious_injuries  \
152    CESSNA   340                  12.0                     0.0   
114    CESSNA   172                  12.0                    10.0   
57      BEECH   A36                   7.0                     0.0   
369     PIPER  PA28                   7.0                     4.0   
122    CESSNA  172N                   7.0                     0.0   
392  ROBINSON   R44                   7.0                     3.0   
373     PIPER  PA34                   6.0                     0.0   
155    CESSNA  414A                   6.0                     0.0   
154    CESSNA   414                   6.0                     0.0   
69       BELL   206                   5.0                     3.0   

     total_minor_injuries  
152                   1.0  
114                   4.0  
57                    1.0  
369                   3.0  
122                   0.0  
392                   2.0  
373                   0.0  
155                   2.0  
154                   0.0  
69                    0.0

In [102]:
#aicraft_damage after accident on the bottom 10 makes
damage_analysis = (
    used_columns
    .groupby(['make', 'aircraft_damage'])
    .size()
    .reset_index(name='Total_Accidents')
    .sort_values('Total_Accidents', ascending=False)
)
damage_analysis.head(10)

make aircraft_damage  Total_Accidents
42         CESSNA     Substantial              169
155         PIPER     Substantial               87
40         CESSNA       Destroyed               31
26          BEECH     Substantial               30
28           BELL     Substantial               23
168      ROBINSON     Substantial               17
153         PIPER       Destroyed               16
25          BEECH       Destroyed               14
46       CHAMPION     Substantial                8
3    AEROSPATIALE     Substantial                7

In [92]:
#makes  with least accidents
bottom_10_make = (
    used_columns['make']
    .value_counts()
    .tail(10)
)
bottom_10_make

make
SIAI-MARCHETTI                1
GROB AIRCRAFT AG              1
DIAMOND AIRCRAFT IND GMBH     1
EVOLUTION AIR LLC             1
PAUL R SEIPT                  1
LEONARDO                      1
BOROM MARCUS P                1
DOSHIER WILBERT A             1
KELLEY DERRICK/LEZA           1
WACO CLASSIC AIRCRAFT CORP    1
Name: count, dtype: int64

In [93]:
#model with most accidents
make_model_accidents.tail(10)

make          model  Total_Accidents
191  CHRISTEN INDUSTRIES INC            A-1                1
189                 CHAMPION          7GCAA                1
188                 CHAMPION           7ECA                1
187             CESSNA ECTOR           305A                1
186       CESSNA AIRCRAFT CO           E162                1
185          CESSNA AIRCRAFT          T210L                1
184                   CESSNA          U206F                1
183                   CESSNA          U206E                1
182                   CESSNA          U206C                1
461                   ZENITH  ZODIAC CH 650                1

In [104]:
#aicraft_damage after accident on the bottom 10 makes
damage_analysis = (
    used_columns
    .groupby(['make', 'aircraft_damage'])
    .size()
    .reset_index(name='Total_Accidents')
    .sort_values('Total_Accidents', ascending=False)
)
damage_analysis.tail(10)

make aircraft_damage  Total_Accidents
104                  HILLER     Substantial                1
105               HITCHCOCK     Substantial                1
106         HOFFMAN DAVID J     Substantial                1
107                    HOOD     Substantial                1
108           HUBER HARTMUT     Substantial                1
109                  HUGHES       Destroyed                1
39                   CASHEN     Substantial                1
111  HUGHES HELICOPTERS INC     Substantial                1
112     JAVRON EXPERIMENTAL     Substantial                1
136          MURRAY FRANK H     Substantial                1

In [106]:
#level of fatality after accident based on make 
fatality_analysis = (
    used_columns
    .groupby(['make', 'model'])
    .agg({
        'total_fatal_injuries': 'sum',
        'total_serious_injuries': 'sum',
        'total_minor_injuries': 'sum'
    })
    .reset_index()
    .sort_values('total_fatal_injuries', ascending=False)
)
fatality_analysis.tail(10)

make          model  total_fatal_injuries  \
200  COSTRUZIONI AERONAUTICHE TECNA          P2006                   0.0   
199   CONSOLIDATED AERONAUTICS INC.  LAKE LA-4-200                   0.0   
197                          CONDOR         CONDOR                   0.0   
196           COLUMBIA AIRCRAFT MFG     LC41 550FG                   0.0   
194              CIRRUS DESIGN CORP           SR22                   0.0   
191         CHRISTEN INDUSTRIES INC            A-1                   0.0   
190                        CHAMPION          7GCBC                   0.0   
188                        CHAMPION           7ECA                   0.0   
187                    CESSNA ECTOR           305A                   0.0   
461                          ZENITH  ZODIAC CH 650                   0.0   

     total_serious_injuries  total_minor_injuries  
200                     0.0                   0.0  
199                     0.0                   0.0  
197                     0.0                   0.0  
196                     0.0                   0.0  
194                     2.0                   0.0  
191                     2.0                   0.0  
190                     1.0                   0.0  
188                     0.0                   0.0  
187                     0.0                   0.0  
461                     0.0                   0.0

- From the above date, we conclude that CESSNA and it's models are the ones leading in flight accidents for the past 20 years of which most of the accidents had fatal injures. However, a good number of planes were substantial.

- Foster also has the least accidents having 1 accident in the past 20 years hence being the best. The condition of all airplanes after accident were substantial apart from HUGHES which was destroyed. No one was injured during the accidents.

In [ ]:
#find most frequent weather condition during accidents
weather_conditions = (
    used_columns
    .groupby('weather_condition')
    .size()
    .reset_index(name='Total_Accidents')
    .sort_values('Total_Accidents', ascending=False)
)

weather_conditions.tail(10)

weather_condition  Total_Accidents
2               VMC              658
0               IMC               43
1               Unk               12